internal/config/viper_store.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2025 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate go tool go.uber.org/mock/mockgen -destination=./mocks.go -package=config github.com/mongodb/mongodb-atlas-cli/atlascli/internal/config ConfigStore
+
+package config
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"slices"
+	"sort"
+	"strings"
+
+	"github.com/pelletier/go-toml"
+	"github.com/spf13/afero"
+	"github.com/spf13/viper"
+)
+
+type ViperConfigStore struct {
+	viper     viper.Viper
+	configDir string
+	fs        afero.Fs
+}
+
+// / ViperConfigStore specific methods
```

Comment:
[nit] feel free to address in a follow up
```suggestion
// ViperConfigStore specific methods
```

Reply:
Fixed